Userland/Libraries/LibWeb/Fetch/Infrastructure/HTTP/Bodies.cpp

```diff
@@ -51,6 +52,44 @@ JS::NonnullGCPtr<WebIDL::Promise> Body::fully_read_as_promise() const
     return WebIDL::create_rejected_promise(realm, JS::InternalError::create(realm, "Reading body isn't fully implemented"sv).release_allocated_value_but_fixme_should_propagate_errors());
 }
 
+// https://fetch.spec.whatwg.org/#body-fully-read
+void Body::fully_read(Web::Fetch::Infrastructure::Body::ProcessBodyCallback process_body, Web::Fetch::Infrastructure::Body::ProcessBodyErrorCallback process_body_error, TaskDestination task_destination) const
+{
+    // FIXME: 1. If taskDestination is null, then set taskDestination to the result of starting a new parallel queue.
+    // FIXME: Handle 'parallel queue' task destination
+    VERIFY(!task_destination.has<Empty>());
+    auto task_destination_object = task_destination.get<JS::NonnullGCPtr<JS::Object>>();
+
+    // 2. Let promise be the result of fully reading body as promise given body.
```

Comment:
Step 2 is `Let successSteps given a byte sequence bytes be to queue a fetch task to run processBody given bytes, with taskDestination.` in the spec. In fact, `fully-reading-body-as-promise` seems to be gone entirely. Please at least state what revision of the spec this is based on :)